halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/RequestGenerateService.java

```diff
@@ -54,17 +59,15 @@ public void prepare(MultipartHttpServletRequest request) throws IOException {
     request.getFileMap().entrySet().stream()
         .forEach(
             et -> {
-              if (!et.getKey().equals(CONFIG_KEY)) {
-                File targetFile = getTargetFile(deploymentConfiguration, et.getKey());
-                if (targetFile != null) {
-                  targetFile.getParentFile().mkdirs();
-                  try {
-                    FileOutputStream outStream = new FileOutputStream(targetFile);
-                    outStream.write(et.getValue().getBytes());
-                  } catch (IOException e) {
-                    throw new RuntimeException(e);
-                  }
-                }
+              if (et.getKey().equals(CONFIG_KEY)) {
+                return;
+              }
+              if (et.getKey().startsWith(LOCAL_FILE_PREFIX)) {
+                // write all local files in hal config root
+                writeFile("", et.getKey(), et.getValue());
+              } else {
+                String filePath = et.getKey().replaceAll("__", File.separator);
```

Comment:
How about `SERVICE_SETTINGS_KEY`?

Reply:
I see that service settings files are processed in `writeServiceSettings`, this loop processes profile files and relative resource files